src/main/scala/temple/builder/ServerBuilder.scala

```diff
@@ -0,0 +1,74 @@
+package temple.builder
+
+import temple.ast.AttributeType.ForeignKey
+import temple.ast.Metadata.{ServiceEnumerable, ServiceLanguage}
+import temple.ast.{Attribute, AttributeType, Metadata, ServiceBlock}
+import temple.builder.project.LanguageConfig.GoLanguageConfig
+import temple.builder.project.ProjectConfig
+import temple.detail.LanguageDetail
+import temple.detail.LanguageDetail.GoLanguageDetail
+import temple.generate.CRUD._
+import temple.generate.database.{PostgresContext, PostgresGenerator, PreparedType}
+import temple.generate.server.{CreatedByAttribute, IDAttribute, ServiceRoot}
+
+import scala.collection.immutable.ListMap
+
+object ServerBuilder {
+
+  def buildServiceRoot(
+    serviceName: String,
+    serviceBlock: ServiceBlock,
+    port: Int,
+    endpoints: Set[CRUD],
+    detail: LanguageDetail,
+  ): ServiceRoot = {
+    val language = serviceBlock.lookupMetadata[ServiceLanguage].getOrElse(ProjectConfig.defaultLanguage)
+    val languageConfig = language match {
+      case ServiceLanguage.Go => GoLanguageConfig
+    }
+
+    val createdBy: CreatedByAttribute = serviceBlock.lookupMetadata[ServiceEnumerable] match {
+      case Some(ServiceEnumerable(true)) =>
+        CreatedByAttribute.EnumerateByCreator(
+          languageConfig.createdByInputName,
+          languageConfig.createdByName,
+          languageConfig.createdByAttributeType,
+        )
+      case Some(ServiceEnumerable(false)) =>
+        CreatedByAttribute.EnumerateByAll(
+          languageConfig.createdByInputName,
+          languageConfig.createdByName,
+          languageConfig.createdByAttributeType,
+        )
+      case None => CreatedByAttribute.None
+    }
+
+    val iDAttribute = IDAttribute("id", AttributeType.UUIDType)
+
+    implicit val context
+      : PostgresContext = PostgresContext(PreparedType.DollarNumbers) //TODO: Does this need to change with server lang?
+    val queries: ListMap[CRUD, String] =
+      DatabaseBuilder
+        .buildQuery(serviceName, serviceBlock, endpoints, iDAttribute, createdBy)
+        .map { case (crud, statement) => crud -> PostgresGenerator.generate(statement) }
+
+    val moduleName: String = detail match {
```

Comment:
This also feels quite Go - should be inside a switch?

Reply:
Well the moduleName needs to be defined in every ServiceBlock anyway - so it needs to be set to something. And the detail matching checks for a GoDetail which should only happen when the server lang is go - though I could put an extra check here if you think that's needed

Reply:
Ah yes I remember this now actually - we agreed to take all the parameters for every `Service`. No worries, leave it as is